src/main/java/org/opensearch/security/auditlog/sink/AbstractInternalOpenSearchSink.java

```diff
@@ -44,13 +52,29 @@ public AbstractInternalOpenSearchSink(
         this.clientProvider = clientProvider;
         this.threadPool = threadPool;
         this.storeOpType = storeOpType;
+        this.lastUsedIndexName = null;
     }
 
     @Override
     public void close() throws IOException {
 
     }
 
+    private boolean createIndexIfAbsent(String indexName) {
+        if (Objects.equals(indexName, lastUsedIndexName)) return true;
+
+        try {
+            final CreateIndexRequest createIndexRequest = new CreateIndexRequest(indexName).settings(indexSettings);
+            final boolean ok = clientProvider.admin().indices().create(createIndexRequest).actionGet().isAcknowledged();
+            lastUsedIndexName = indexName;
+            log.info("Index {} created?: {}", indexName, ok);
+            return ok;
+        } catch (ResourceAlreadyExistsException resourceAlreadyExistsException) {
+            log.info("Index {} already exists", indexName);
```

Comment:
Shouldn't `lastUsedIndexName` be set here?

What happens in the case of node reboot and the audit log index exists? `lastUsedIndexName` would be null when the node starts and shouldn't this `catch` clause be used to populate it if it is null and the exist already exists?

Reply:
FYI I experimented around with passing the `clusterService` to the internal opensearch sinks which can be used to check if an index exists without having to issue an Exists request. Take a look at the change in https://github.com/cwperks/security/pull/44. That change also shows an example of how to use the integration test framework in this repo to write an integration test for this case.